src/pages/index.js

```diff
@@ -0,0 +1,375 @@
+const deleteImageCancelButton = document.querySelector("#cancel-delete-btn");
+const editProfileSubmitButton = document.querySelector(
+  "#edit-profile-submit-btn"
+);
+
+// Card Template and List Elements
+const cardTemplate = document.querySelector("#card-template");
+const cardList = document.querySelector(".cards__list");
+
+// Event Listeners
+if (settingsButton) {
+  settingsButton.addEventListener("click", () => {
+    editModalNameInput.value = profileName.textContent;
+    editModalDescriptionInput.value = profileDescription.textContent;
+    resetValidation(
+      editFormElement,
+      [editModalNameInput, editModalDescriptionInput],
+      settings
+    );
+    openModal(editModal);
+  });
+}
+if (editModalExitButton) {
+  editModalExitButton.addEventListener("click", () => {
+    closeModal(editModal);
+  });
+}
+if (editFormElement) {
+  editFormElement.addEventListener("submit", (evt) =>
+    editProfile(evt, settings)
+  );
+}
+if (postButton) {
+  postButton.addEventListener("click", () => {
+    openModal(postModal);
+  });
+}
+if (editAvatarFormElement) {
+  editAvatarFormElement.addEventListener("submit", (evt) => {
+    editAvatar(evt);
+  });
+}
+if (editAvatarButton) {
+  editAvatarButton.addEventListener("click", () => {
+    openModal(editAvatarModal);
+  });
+}
+if (editAvatarExitButton) {
+  editAvatarExitButton.addEventListener("click", () => {
+    closeModal(editAvatarModal);
+  });
+}
+if (postModalExitButton) {
+  postModalExitButton.addEventListener("click", () => {
+    closeModal(postModal);
+  });
+}
+if (postFormElement) {
+  postFormElement.addEventListener("submit", (evt) => postCard(evt, settings));
+}
+if (viewImageModalCloseButton) {
+  viewImageModalCloseButton.addEventListener("click", () => {
+    closeModal(viewImageModal);
+  });
+}
+if (deleteImageExitButton) {
+  deleteImageExitButton.addEventListener("click", () => {
+    closeModal(deleteImageModal);
+  });
+}
+
+// Functions
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    closeModal(document.querySelector(".modal_opened"));
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+function editProfile(evt) {
+  evt.preventDefault();
+
+  editProfileSubmitButton.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      editProfileSubmitButton.textContent = "Submit";
+    });
+}
+
+function editAvatar(evt) {
+  evt.preventDefault();
```

Comment:
When you submit the avatar form, you should change the button text as well: `"Saving..."`